mypy/errors.py

```diff
@@ -153,13 +158,14 @@ class Errors:
     target_module = None  # type: Optional[str]
     scope = None  # type: Optional[Scope]
 
-    def __init__(self,
-                 show_error_context: bool = False,
-                 show_column_numbers: bool = False,
-                 show_error_codes: bool = False) -> None:
-        self.show_error_context = show_error_context
-        self.show_column_numbers = show_column_numbers
-        self.show_error_codes = show_error_codes
+    def __init__(self, fscache: Optional[FileSystemCache] = None,
+                 options: Optional[Options] = None) -> None:
```

Comment:
This adds a big dependency, which we previously avoided by providing the required information explicitly. It would arguably be cleaner if you'd just add the required options as arguments here instead of passing the whole `Options` object.

Reply:
OK, I will split this back.